### ISSUE TITLE
NullPointerException using `file:%workspace%/registry` in registry URI on Windows

### ISSUE THREAD

**Issue by avdv** [bazelbuild/bazel#20015]
### Description of the bug:

Using a registry URI such as `file:%workspace%/../registry` leads to a NPE on Windows:
```
java.lang.RuntimeException: Unrecoverable error while evaluating node 'RepoSpecKey{moduleKey=rules_haskell_nix@0.16, registryUrl=file:%workspace%/../registry}' (requested by nodes 'com.google.devtools.build.lib.bazel.bzlmod.BazelModuleResolutionValue$$Lambda$381/0x0000000100508840@549d485d')
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:633)
        at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:365)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException
        at com.google.devtools.build.lib.bazel.bzlmod.IndexRegistry.createLocalPathRepoSpec(IndexRegistry.java:213)
        at com.google.devtools.build.lib.bazel.bzlmod.IndexRegistry.getRepoSpec(IndexRegistry.java:177)
        at com.google.devtools.build.lib.bazel.bzlmod.RepoSpecFunction.compute(RepoSpecFunction.java:51)
        at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:562)
```

### Which category does this issue belong to?

Loading API

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run bazel with the `--registry=file:%workspace%/registry --enable_bzlmod` in a module on Windows.

### Which operating system are you running Bazel on?

Windows

### What is the output of `bazel info release`?

release 6.4.0

### Any other information, logs, or outputs that you want to share?

This is caused by the Windows workspace path just being inserted into that string. 

The resulting URI looks like this: `file:c:/path/to/workspace/registry`

This is not a valid file URI and thus `uri.getPath()` returns null which causes the exception.

The workaround is to use triple slashes: `--registry=file:///%workspace%/registry` (this leads to the correct path on Windows, and to a path starting with `//` on non-Windows).

**Comment by meteorcloudy** [comment]
@bazel-io fork 7.0.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.0.0 RC5](https://releases.bazel.build/7.0.0/rc5/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!
